foursight_core/app_utils.py

```diff
@@ -334,6 +547,175 @@ def trim_output(cls, output, max_size=100000):
             return cls.TRIM_ERR_OUTPUT
         return output
 
+    def sorted_dict(self, dictionary: dict) -> dict:
+        result = {}
+        for key in sorted(dict(dictionary).keys(), key=lambda key: key.lower()):
+            result[key] = dictionary[key]
+        return result
```

Comment:
Untested, but I think this is pretty much the same thing and more "pythonic":

```suggestion
    def sorted_dict(self, dictionary: dict) -> dict:
        dictionary = dict(dictionary)
        return {result[key]: dictionary[key]
                for key in sorted(dictionary.keys(), key=lambda key: key.lower())}
```

This also needs a doc string that explains what kinds of arguments it takes. It's odd because in the original form it seems to take dictionary-like things of some kind that can be coerced to dicts and that can be indexed like dicts. I added the `dictionary = dict(dictionary)` on the first line so it could be more confident later it's a dict. e.g., a list of tuples (pairs) can be coerced to a dictionary but cannot be random-accessed by key, yet will work here if you do the coercion first that you're planning to do anyway just to throw away.